src/bld/linuxCfg.js

```diff
@@ -6,30 +6,48 @@ import { log } from "../log.js";
  * Generates a Desktop Entry file for Linux
  * https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html
  *
- * @param  {object}    pkg     srcDir's package.json as JSON
- * @param  {string}    outDir  directory which stores build artifacts
+ * @param  {object}    app     Multi platform configuration options
+ * @param  {string}    outDir  Directory which stores build artifacts
  * @return {undefined}
  */
-export const setLinuxConfig = async (pkg, outDir) => {
+export const setLinuxConfig = async (app, outDir) => {
   let desktopEntryFile = {
     Type: "Application",
-    Name: pkg.name,
-    Exec: pkg.name,
+    Version: "1.5",
+    Name: app.name,
+    GenericName: app.genericName,
+    NoDisplay: app.noDisplay,
+    Comment: app.comment,
+    Icon: app.icon,
+    Hidden: app.hidden,
+    OnlyShowIn: app.onlyShowIn,
+    NotShowIn: app.notShowIn,
+    DBusActivatable: app.dBusActivatable,
+    TryExec: app.tryExec,
+    Exec: app.name,
+    Path: app.path,
+    Terminal: app.terminal,
+    Actions: app.actions,
+    MimeType: app.mimeType,
+    Categories: app.categories,
+    Implements: app.implements,
+    Keywords: app.keywords,
+    StartupNotify: app.startupNotify,
+    StartupWMClass: app.startupWMClass,
+    PrefersNonDefaultGPU: app.prefersNonDefaultGPU,
+    SingleMainWindow: app.singleMainWindow,
   };
-  await rename(`${outDir}/nw`, `${outDir}/${pkg.name}`);
-  if (typeof pkg.nwbuild?.linuxCfg === "object") {
-    Object.keys(pkg.nwbuild.linuxCfg).forEach((key) => {
-      if (key !== "Type") {
-        desktopEntryFile[key] = pkg.nwbuild.linuxCfg[key];
-      }
-    });
-  }
+
+  await rename(`${outDir}/nw`, `${outDir}/${app.name}`);
+
   let fileContent = `[Desktop Entry]\n`;
   Object.keys(desktopEntryFile).forEach((key) => {
-    fileContent += `${key}=${desktopEntryFile[key]}\n`;
-    log.debug(`Add ${key}=${desktopEntryFile[key]} to Desktop Entry File`);
+    if (desktopEntryFile[key] !== undefined) {
+      fileContent += `${key}=${desktopEntryFile[key]}\n`;
+      log.debug(`Add ${key}=${desktopEntryFile[key]} to Desktop Entry File`);
+    }
```

Comment:
This PR make a lot of changes to this file. Most of the file's code is focused on creating a desktop shortcut. The NW-Utils GitHub organization already has a library that handles that:

* https://github.com/nwutils/create-desktop-shortcuts

It additionally sets the chmod on the shortcut to allow the shortcut to be executed. Since that library is already fully unit tested, we should pull it in here to reduce the amount of code in this file and shift the burden of testing this logic over to a codebase scoped to handle just that feature.

Also there are several parameters listed in this PR that are not recognized in all Linux distros. So they are not included in the `create-desktop-shortcut` API. I'm not sure if any would actually be relevant to NW-Builder users. But if so we should probably add them to the `create-desktop-shortcuts` library anyway.

Reply:
Good to know! I'll have a look at it soon.

Reply:
Can be handled in a future PR as well, I don't want to slow progress on this project.

Reply:
I have created a separate issue for it.